pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -399,7 +410,7 @@ func (e *EndpointController) syncService(key string) error {
 	subsets = endpoints.RepackSubsets(subsets)
 
 	// See if there's actually an update here.
-	currentEndpoints, err := e.client.Core().Endpoints(service.Namespace).Get(service.Name, metav1.GetOptions{})
+	currentEndpoints, err := e.endpointsLister.Endpoints(service.Namespace).Get(service.Name)
```

Comment:
+1. You need to do deep copy in line 432 (in case when you didn't enter `if` in line 415).